src/main/kotlin/no/nav/syfo/melding/MeldingService.kt

```diff
@@ -21,10 +25,17 @@ class MeldingService(
         // TODO: legg til i joark-cronjob og send på kafka her
     }
 
-    fun getMeldingerTilBehandler(personIdent: PersonIdent): List<MeldingTilBehandler> {
-        // TODO: Her ville vi vel gruppere det på en eller annen måte for ulike samtaler/dialoger
-        return database.getMeldingerForArbeidstaker(personIdent)
-            .filter { !it.innkommende } // Foreløpig bare ta MeldingTilBehandler-meldingene
-            .map { it.toMeldingTilBehandler() }
+    fun getConversationMeldingerMap(personIdent: PersonIdent): Map<UUID, List<Melding>> {
+        val meldinger = database.getMeldingerForArbeidstaker(personIdent)
+        return meldinger.groupBy(
+            keySelector = { it.conversationRef },
+            valueTransform = {
+                if (it.innkommende) {
+                    it.toMeldingFraBehandler().toMelding(it.behandlerRef!!)
```

Comment:
Denne `it.behandlerRef!!` vil jo feile med en gang vi legger `MeldingFraBehandler`-meldinger i databasen. Foreløpig er det bare `MeldingTilBehandler`-meldinger i testene.